topostats/grainstats.py

```diff
@@ -0,0 +1,653 @@
+
+        Parameters
+        ----------
+        edges : list
+            A python list of coordinates that make up the edges of the grain.
+
+        Returns
+        -------
+        hull : list
+            A list containing coordinates of the points in the hull.
+        hull_indices : list
+            A list containing the hull points indices inside the edges list. In other words, this provides a way to find
+            the points from the hull inside the edges list that was passed.
+        simplices : list
+            A  list of tuples, each tuple representing a simplex of the convex hull. These simplices are sorted such
+            that they follow each other in counterclockwise order.
+        """
+        # FIXME : Make this an isolated method
+        # Find a point guaranteed to be on the hull. I find the bottom most point(s) and sort by x-position.
+        min_y_index = None
+        for index, point in enumerate(edges):
+            if min_y_index == None or point[1] < edges[min_y_index][1]:
+                min_y_index = index
+            if point[1] == edges[min_y_index][1] and point[0] < edges[min_y_index][0]:
+                min_y_index = index
+        self.start_point = edges[min_y_index]
+        # This does the same thing, but as a separate method and with Numpy Array rather than a lsit
+        # self.get_start_point(edges)
+        # Sort the points
+        points_sorted_by_angle = self.sort_points(edges)
+
+        # Remove starting point from the list so it's not added more than once to the hull
+        start_point_index = points_sorted_by_angle.index(self.start_point)
+        del points_sorted_by_angle[start_point_index]
+        # Add start point and the first point sorted by angle. Both of these points will always be on the hull.
+        hull = [self.start_point, points_sorted_by_angle[0]]
+
+        # Iterate through each point, checking if this point would cause a clockwise rotation if added to the hull, and
+        # if so, backtracking.
+        for index, point in enumerate(points_sorted_by_angle[1:]):
+            # Determine if the proposed point demands a clockwise rotation
+            while self.is_clockwise(hull[-2], hull[-1], point) == True:
+                # Delete the failed point
+                del hull[-1]
+                if len(hull) < 2:
+                    break
+            # The point does not immediately cause a clockwise rotation.
+            hull.append(point)
+
+        # Get hull indices from original points array
+        hull_indices = []
+        for point in hull:
+            hull_indices.append(edges.index(point))
+
+        # Create simplices from the hull points
+        simplices = []
+        for i in range(len(hull_indices)):
+            simplices.append((hull_indices[i - 1], hull_indices[i]))
+
+        return hull, hull_indices, simplices
+
+    def plot(self, edges: list, convex_hull: list = None, file_path: Path = None):
```

Comment:
Maybe plotting should be seperated from analysis code?

Reply:
That was my hacky solution to not moving the plotting out to `topostats.plottingfuncs` as @SylviaWhittle said that its mainly code she was using whilst working through writing this class. Will investigate methods of having plots created on entering debugging and agree that all plotting functions should be in their own sub-module.